api/hypershift/v1beta1/azure_types.go

```diff
@@ -1,144 +1,177 @@
 	// HostedCluster.Spec.Platform.Azure.SubscriptionID.
+	// subnetID is immutable once set.
+	// TODO: What is the valid character set for this field? What about minimum and maximum lengths?
 	//
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="SubnetID is immutable"
 	// +kubebuilder:validation:Required
 	SubnetID string `json:"subnetID"`
 
-	// Diagnostics specifies the diagnostics settings for a virtual machine.
+	// diagnostics specifies the diagnostics settings for a virtual machine.
 	// If not specified, then Boot diagnostics will be disabled.
 	// +optional
 	Diagnostics *Diagnostics `json:"diagnostics,omitempty"`
 
-	// MachineIdentityID is a user-assigned identity assigned to the VMs used to authenticate with Azure services. This
-	// field is expected to exist under the same resource group as HostedCluster.Spec.Platform.Azure.ResourceGroupName. This
+	// machineIdentityID is a user-assigned identity assigned to the VMs used to authenticate with Azure services. The
+	// identify is expected to exist under the same resource group as HostedCluster.Spec.Platform.Azure.ResourceGroupName. This
 	// user assigned identity is expected to have the Contributor role assigned to it and scoped to the resource group
 	// under HostedCluster.Spec.Platform.Azure.ResourceGroupName.
 	//
 	// If this field is not supplied, the Service Principal credentials will be written to a file on the disk of each VM
 	// in order to be accessible by the cloud provider; the aforementioned credentials provided are the same ones as
 	// HostedCluster.Spec.Platform.Azure.Credentials. However, this is less secure than using a managed identity.
 	//
+	// TODO: What is the valid character set for this field? What about minimum and maximum lengths?
+	//
 	// +optional
 	MachineIdentityID string `json:"machineIdentityID,omitempty"`
 }
 
-// AzureVMImage represents the different types of image data that can be provided for an Azure VM.
+// AzureVMImage represents the different types of boot image sources that can be provided for an Azure VM.
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'ImageID' ?  has(self.imageID) : !has(self.imageID)",message="imageID is required when type is ImageID, and forbidden otherwise"
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'AzureMarketplace' ?  has(self.azureMarketplace) : !has(self.azureMarketplace)",message="azureMarketplace is required when type is RequiredMember, and forbidden otherwise"
 // +union
 type AzureVMImage struct {
-	// Type is the type of image data that will be provided to the Azure VM. This can be either "ImageID" or
-	// "AzureMarketplace".
+	// type is the type of image data that will be provided to the Azure VM.
+	// Valid values are "ImageID" and "AzureMarketplace".
+	// ImageID means ... TODO
+	// AzureMarketplace means ... TODO
```

Comment:
Help the user to choose the correct value by telling them what each one means

Reply:
source of info for azure marketplace images - https://techcommunity.microsoft.com/t5/marketplace-blog/exploring-virtual-machine-offers-in-the-azure-marketplace/ba-p/3779854